data_structures/arrays/permutations.py

```diff
@@ -0,0 +1,31 @@
+from itertools import permutations
+
+
+# All permutations:
+def permute(nums: list[int]) -> list[list[int]]:
+    """
+    >>> from itertools import permutations
+    >>> numbers= [1,2,3]
+    >>> all(list(nums) in permute(numbers) for nums in permutations(numbers))
+    True
+    """
+    result = []
+    if len(nums) == 1:
+        return [nums.copy()]
+    for _ in range(len(nums)):
+        n = nums.pop(0)
+        permutations = permute(nums)
+        for perm in permutations:
+            perm.append(n)
+        result.extend(permutations)
+        nums.append(n)
```

Comment:
You're mutating the input in-place! That is a very bad idea. The function is returning the result, so why are you mutating the input?